src/vos/vos_tree.c

```diff
@@ -962,6 +962,34 @@ key_tree_prepare(struct vos_object *obj, daos_epoch_t epoch,
 
 		kbund.kb_epoch	= DAOS_EPOCH_MAX;
 		rbund.rb_iov	= key;
+		if (tclass == VOS_BTR_DKEY) {
+			struct dtx_handle	*dth = vos_dth_get();
+
+			/* Dkey is the determining factor for on which target
+			 * the data will be stored: the akeys and SV/EV tree
+			 * records belong to the same dkey will be stored on
+			 * the same target. But for different dkeys belong to
+			 * the same redundancy group (for the same replicated
+			 * object), the leaders replica for these different
+			 * dkeys may reside on different targets in the same
+			 * redundancy group.
+			 *
+			 * For VOS_ITER_DKEY operation, if some new dkeys which
+			 * DTXs are committable but related leader replicas are
+			 * on different targets, then in spite of list dkeys on
+			 * which replica, the client will get -DER_INPROGRESS
+			 * and has to retry the list_dkey RPC again and again.
```

Comment:
just to confirm that it possible retry list_dkey RPC for some times, but should not be forever right?

is it possible for another options that don't do synchronize commit when key create, but do two other changes:
1) when server serves list_dkey, for -DER_INPROGRESS case it also returns one (the first) conflicted DTX
2) when client retry list_dkey RPC to leader, it carries a flag ("FORCE_COMMIT" or sth else) and the collected conflicted DTX array, when server do synchronous commit for those DTXs and reply to client;
3) based on 2's reply, client maybe can get its result or retry...

seems it is similar as EC fetch's EINPROGRESS handling